src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,102 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LENGTH = 10;
+    private static final double MULTIPLIER = 1.5;
+    private T[] currentArray;
+    private int size;
+
+    public ArrayList() {
+        currentArray = (T[]) new Object[DEFAULT_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == currentArray.length) {
+            increaseLength();
+        }
```

Comment:
I think it is redundant here, since we have it other `add()` method

Reply:
If i remove this size check - there is possible out of bounds exception - in case if currentArray is already have 10 elements.

Reply:
@RandomEastEcho I think @karma-o meant that you can reuse your add(T value, int index) method here